crates/outline/src/outline.rs

```diff
@@ -241,48 +245,80 @@ impl PickerDelegate for OutlineViewDelegate {
                 })
                 .collect();
 
-            let (buffer, cursor_offset) = self.active_editor.update(cx, |editor, cx| {
-                let buffer = editor.buffer().read(cx).snapshot(cx);
-                let cursor_offset = editor.selections.newest::<usize>(cx).head();
-                (buffer, cursor_offset)
-            });
-            selected_index = self
+            let selected_index = self
                 .outline
                 .items
                 .iter()
                 .enumerate()
                 .map(|(ix, item)| {
                     let range = item.range.to_offset(&buffer);
-                    let distance_to_closest_endpoint = cmp::min(
-                        (range.start as isize - cursor_offset as isize).abs(),
-                        (range.end as isize - cursor_offset as isize).abs(),
-                    );
-                    let depth = if range.contains(&cursor_offset) {
-                        Some(item.depth)
-                    } else {
-                        None
-                    };
+                    let distance_to_closest_endpoint =
+                        get_distance_to_closest_endpoint(&range, cursor_offset);
+                    let depth = Some(item.depth);
                     (ix, depth, distance_to_closest_endpoint)
                 })
                 .max_by_key(|(_, depth, distance)| (*depth, Reverse(*distance)))
                 .map(|(ix, _, _)| ix)
                 .unwrap_or(0);
-        } else {
-            self.matches = smol::block_on(
-                self.outline
-                    .search(&query, cx.background_executor().clone()),
-            );
-            selected_index = self
-                .matches
-                .iter()
-                .enumerate()
-                .max_by_key(|(_, m)| OrderedFloat(m.score))
-                .map(|(ix, _)| ix)
-                .unwrap_or(0);
+
+            self.last_query = query;
+            self.set_selected_index(selected_index, false, cx);
+            return Task::ready(());
         }
-        self.last_query = query;
-        self.set_selected_index(selected_index, !self.last_query.is_empty(), cx);
-        Task::ready(())
+
+        let outline = self.outline.clone();
```

Comment:
Is this the only reason the `Clone` derive was added?
I think we could go with `outline_view.read_with`/`outline_view.update` to get this property out without any API additions?

Just feels somewhat odd to make this particular change for the test only.

`cx.background_executor` should be also available from the `cx` inside `cx.spawn`, so could be also inlined?